google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -2474,6 +2466,8 @@ A toPrimitiveArray(int columnIndex) {
   }
 
   private static class Int64Array extends PrimitiveArray<Long, long[]> {
```

Comment:
Making all these arrays serializable in this way ties us very tightly to how they are implemented currently and makes it harder to change. For eg we might decide that we should change BooleanArray also to extend PrimitiveArray or maybe we decide that we should just simplify Int64Array and do away with all this business of bitsets. Or we might do away with all these classes entirely and just return an ArrayList<Integer>. It might be better to implement a custom serialization for GrpcStruct possibly using a serialization proxy as described in effective java:
https://books.google.com/books?id=ka2VUBqHiWkC&lpg=PA312&ots=yZLjMhu2M3&dq=serialization%20proxy&pg=PA312#v=onepage&q=serialization%20proxy&f=false